resources/asciidoctor/lib/elastic_compat_preprocessor/extension.rb

```diff
@@ -119,77 +119,115 @@ class ElasticCompatPreprocessor < Asciidoctor::Extensions::Preprocessor
       elsif line == '--'
-        lines = self.lines
-        lines.shift
-        while Asciidoctor::AttributeEntryRx =~ (check_line = lines.shift)
-        end
-        return line unless check_line == '--'
-
-        @in_attribute_only_block = true
-        line.clear
+        process_start_block line
+      elsif (match = INCLUDE_TAGGED_DIRECTIVE_RX.match line)
+        process_include_tagged line, match[1], match[2]
       else
-        line = super
-        return if line.nil?
-
-        SOURCE_WITH_SUBS_RX.match(line) do |m|
-          # AsciiDoc would automatically add `subs` to every source block but
-          # Asciidoctor does not and we have thousands of blocks that rely on
-          # this behavior.
-          old_subs = m[1]
-          line.sub! "subs=\"#{old_subs}\"", "subs=\"#{old_subs},callouts\"" unless old_subs.include? 'callouts'
-        end
-        if CODE_BLOCK_RX =~ line
-          if @code_block_start
-            if line != @code_block_start
-              line.replace(@code_block_start)
-              message = "MIGRATION: code block end doesn't match start"
-              logger.warn message_with_context message, source_location: cursor
-            end
-            @code_block_start = nil
-          else
-            @code_block_start = line
+        postprocess super
+      end
+    end
+
+    ##
+    # Handle a line if we're in attribute only block. We are basically a
+    # passthrough in this state, just hunting for the block end. If we hit the
+    # block end we eat the block delimiter because we ate the start delimiter
+    # when entering into the attribute only block.
+    def process_in_attribute_only_block(line)
+      return line unless line == '--'
+
+      @in_attribute_only_block = false
+      line.clear
+    end
+
+    ##
+    # Process a start block when, potentially shifting into the "attribute only"
+    # block state if the block that is starting only contains attributes. If
+    # we enter into that state then we eat the block delimiter to work around
+    # a scoping difference between AsciiDoc and Asciidoctor.
+    def process_start_block(line)
+      lines = self.lines
+      lines.shift
+      while Asciidoctor::AttributeEntryRx =~ (check_line = lines.shift)
+      end
+      return line unless check_line == '--'
+
+      @in_attribute_only_block = true
+      line.clear
+    end
+
+    ##
+    # Process the `include-tagged` directive.
+    def process_include_tagged(line, target, tag)
+      return if preprocess_include_directive(
+        "elastic-include-tagged:#{target}", tag
+      )
+
+      # the line was not a valid include line and we've logged a warning
+      # about it so we should do the asciidoctor standard thing and keep
+      # it intact. This is how we do that.
+      @look_ahead += 1
+      line
+    end
+
+    ##
+    # Process lines after they've been processed by the reader.
+    def postprocess(line)
+      return if line.nil?
```

Comment:
I would write instead `return unless line`

Reply:
:+1:. I keep getting mixed up about empty lines being truthy. In perl it is falsey. Javascript too.

Reply:
Welcome to Rubyland =)